internal/services/environment_wave/api_environment_wave.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package environment_wave
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/microsoft/terraform-provider-power-platform/internal/api"
+	"github.com/microsoft/terraform-provider-power-platform/internal/constants"
+)
+
+type environmentWaveClient struct {
+	Api *api.Client
+}
+
+func newEnvironmentWaveClient(apiClient *api.Client) *environmentWaveClient {
+	return &environmentWaveClient{
+		Api: apiClient,
+	}
+}
+
+func (client *environmentWaveClient) UpdateFeature(ctx context.Context, environmentId string, featureName string) (*FeatureDto, error) {
+	apiUrl := &url.URL{
+		Scheme: constants.HTTPS,
+		Host:   "api.admin.powerplatform.microsoft.com",
+		Path:   fmt.Sprintf("/api/environments/%s/features/%s/enable", environmentId, featureName),
+	}
+
+	_, err := client.Api.Execute(ctx, nil, "POST", apiUrl.String(), nil, nil, []int{http.StatusOK}, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	retryAfter := api.DefaultRetryAfter()
```

Comment:
does this not return a retry after header?

Reply:
no, we have to query state from GET operation